src/libraries/System.Private.CoreLib/src/System/Number.Parsing.cs

```diff
@@ -295,7 +295,7 @@ private static unsafe bool TryParseNumber(ref char* str, char* strEnd, NumberSty
                 // "-Kr 1231.47" is legal but "- 1231.47" is not.
                 if (!IsWhite(ch) || (styles & NumberStyles.AllowLeadingWhite) == 0 || ((state & StateSign) != 0 && ((state & StateCurrency) == 0 && info.NumberNegativePattern != 2)))
                 {
-                    if ((((styles & NumberStyles.AllowLeadingSign) != 0) && (state & StateSign) == 0) && ((next = MatchChars(p, strEnd, info.PositiveSign)) != null || ((next = MatchChars(p, strEnd, info.NegativeSign)) != null && (number.IsNegative = true))))
+                    if ((((styles & NumberStyles.AllowLeadingSign) != 0) && (state & StateSign) == 0) && ((next = MatchChars(p, strEnd, info.PositiveSign)) != null || ((next = MatchNegativeSignChars(p, strEnd, info)) != null && (number.IsNegative = true))))
```

Comment:
BigInteger has an old copy of the parsing code:
https://github.com/dotnet/runtime/blob/master/src/libraries/Common/src/System/Globalization/FormatProvider.Number.cs
There's an issue to merge them again (https://github.com/dotnet/runtime/issues/28657), but until that happens, presumably we need to update that as well?

Reply:
I am seeing this file is used in System.Runtime.Numeric only. it is even weird that this stored in the common folder. I guess that was the file we used in the globalization parsing before and it didn't get chance to be cleaned up. anyway, I'll add the fix to this file too.